addons/webhook/certificates.go

```diff
@@ -0,0 +1,125 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	rsrcs "knative.dev/pkg/webhook/certificates/resources"
+	"path"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"time"
+)
+
+const (
+	ServerKey = "server-key.pem"
+	// ServerCert is the name of the key associated with the secret's public key.
+	ServerCert = "server-cert.pem"
+	// CACert is the name of the key associated with the certificate of the CA for
+	// the keypair.
+	CACert = "ca-cert.pem"
+)
+
+func SetupWebhooks(mgr *manager.Manager, namespace string, vWhCfgNames, mWhCfgNames []string) error {
+
+	//create the secret and write the certificate files based on the existing webhook server config
+	wh := (*mgr).GetWebhookServer()
+	secret, err := UpdateTLSSecret("addons-webhook-tls", wh, namespace)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Rotation could be setup here similar to antrea: https://github.com/antrea-io/antrea/blob/main/pkg/apiserver/certificate/certificate.go#L100

Reply:
Timer based with select is probably better(this is sleep). Also needs to reload webhook http server with new certs.

Reply:
Its starting to look like we need a certificate controller.. Certificate roration coudl be taken care of *and* we need to no only update existing webhook Configurations but also configurations which are created *after* the add on manger is started.

Reply:
I think we will submit rotation of certificates as a follow up pr to keep this pr small and not break current functioanlity